### PR TITLE
Issue #5214 - Servlet HEAD doesn't support content-length over Integer.MAX_VALUE

### LINKED ISSUE #5214: Servlet HEAD doesn't support content-length over Integer.MAX_VALUE
9.4.31.v20200723

(build 11.0.8+10-jvmci-20.2-b03

Windows 10

When doing a HEAD request to retrieve file size using Jetty, anything one byte or more over 2^31 returns Content-Length: 0.

### PR DESCRIPTION
**Pull request by joakime**

### CODE REVIEW

**Review by gregw — COMMENTED** [review]
LGTM, but t hen I wrote half of it.... need another reviewer :)

**Comment by ebremer** [comment]
Thank you genetlemen!  I will retest after release.  Any ETA on that?

**Comment by gregw** [comment]
@ebremer I believe we are doing releases in the next week or two.... but if you could test this before we release, then that helps us get to a release sooner!

**Comment by ebremer** [comment]
@gregw I am building 9.4.x now.  I will let you know how it does.

**Comment by ebremer** [comment]
@gregw if the build is 9.4.32-SNAPSHOT, it's still failing.

**Comment by ebremer** [comment]
@gregw Nevermind.  The PR was not merged yet.  I rebuilt using jetty-9.4.x-5214-head-huge-static and it works fine now on my installation.

File: jetty-webapp/src/test/java/org/eclipse/jetty/webapp/HugeResourceTest.java [hunk 1]
```diff
@@ -244,6 +244,26 @@ public void testDownload(String filename, long expectedSize) throws Exception
...
+    @MethodSource("staticFiles")
+    public void testHead(String filename, long expectedSize) throws Exception
+    {
+        URI destUri = server.getURI().resolve("/" + filename);
+        InputStreamResponseListener responseListener = new InputStreamResponseListener();
+
+        Request request = client.newRequest(destUri)
+            .method(HttpMethod.HEAD);
+        request.send(responseListener);
+        Response response = responseListener.get(5, TimeUnit.SECONDS);
```
Comment (sbordet) [inline jetty-webapp/src/test/java/org/eclipse/jetty/webapp/HugeResourceTest.java:257]: `responseListener.get()` only waits for the headers, so to be sure here I would also do (maybe in a try-with-resources):

`assertEquals(-1, response.getInputStream().read());`

which would guarantee that the HEAD response really came without body, as it should be.

**Review by sbordet — CHANGES_REQUESTED** [review]

File: jetty-webapp/src/test/java/org/eclipse/jetty/webapp/HugeResourceTest.java (hunk 1 above)
Comment (joakime) [reply jetty-webapp/src/test/java/org/eclipse/jetty/webapp/HugeResourceTest.java:257]: Good point.  
But if that fails, does that mean that both jetty server **and** jetty client have bugs then? ;-)

Comment (sbordet) [reply jetty-webapp/src/test/java/org/eclipse/jetty/webapp/HugeResourceTest.java:257]: Does not matter - we will have a failed test and we'll investigate :)

Comment (gregw) [reply jetty-webapp/src/test/java/org/eclipse/jetty/webapp/HugeResourceTest.java:257]: @joakime I'll fix this to get this merged ASAP.

**Review by sbordet — APPROVED** [review]

**Pull request merged**